Makefile

```
@@ -30,3 +30,12 @@ include packaging/Makefile_packaging.mk
 #   git diff $(VERSION)..$(GIT_COMMIT)              \
 #       > ../$@
 #   git add $@
+
+$(NAME)-$(VERSION).tar.gz: $(shell git ls-files :/:)
+	echo Creating $@
+	echo $(basename $@)
+	rm -f $@
+	git submodule update --init --recursive
+	git ls-files --recurse-submodules $(NAME)-$(VERSION) | tar caf $(NAME)-$(VERSION).tar.gz -T-
```

FWIW: This would allow you to use the "spdk" name for the submodule while still getting the correctly-named top directory in the tarball:
```suggestion
	git ls-files --recurse-submodules $(NAME) | tar caf $(NAME)-$(VERSION).tar.gz --transform='s/^$(NAME)/$(NAME)-$(VERSION)/' -T-
```